dbms/src/IO/HDFSCommon.cpp

```diff
@@ -25,6 +26,13 @@ HDFSBuilderPtr createHDFSBuilder(const Poco::URI & uri)
     hdfsBuilderConfSetStr(builder.get(), "input.write.timeout", "60000"); // 1 min
     hdfsBuilderConfSetStr(builder.get(), "input.connect.timeout", "60000"); // 1 min
 
+    std::string userInfo = uri.getUserInfo();
+    if (!userInfo.empty() && userInfo.front() != ':') {
+        char *user = strtok(const_cast<char *> (userInfo.c_str()), ":");
```

Comment:
Need to rewrite in C++.
PS. strtok can damage password field if it contains colon.

Reply:
```
BUGS
       * The strtok() function uses a static buffer while parsing, so it's not thread safe.  Use strtok_r() if this matters to you.
```

Reply:
Rewrite in c++ 
Used  boost::split like in other code examples in repo.
Fixed code styles bugs
NULL -> nullptr for fix clang builds

Reply:
You can just use std::string::find.

Reply:
rewritten from boost to std::string